src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/ngram_fusion.cpp

```diff
@@ -153,7 +166,7 @@ ov::intel_cpu::NgramFusion::NgramFusion() {
             auto then_m = wrap_type<StridedSlice>({padded_tokens_m, wrap_type<Constant>(), then_cropped_shape_m, wrap_type<Constant>()});
 
             // else branch
-            auto else_target_shape_concat_m = wrap_type<Concat>({cropped_shape_m, wrap_type<Constant>()});
+            auto else_target_shape_concat_m = any_input(concat_shape_with_cropped_symbol_m);
```

Comment:
Could you explain why you didn't use Or here as in similar cases above?
<img width="703" alt="{448AD2F1-429A-4A83-AC9D-CBA1C48DB0A6}" src="https://github.com/user-attachments/assets/2d7b4063-7d9e-46a2-a34f-a2ef62660f3a">

Reply:
Is added as in some cases when bias is not added the symbolic optimization sees that Concat is same connectino as shape-of input parameter and model is reduced. 
This optional check just check if instead Concat there is Op which provide output with shape which got `cropped symbol `.

Reply:
do we really need to check Concat in the above cases, isn't it enough to check `cropped symbol` and not use Or?